src/Orleans/Messaging/ProxiedMessageCenter.cs

```diff
@@ -296,16 +307,26 @@ public Message WaitMessage(Message.Categories type, CancellationToken ct)
 #endif
                 return msg;
             }
-            catch (ThreadAbortException tae)
+            catch (ThreadAbortException exc)
             {
                 // Silo may be shutting-down, so downgrade to verbose log
-                logger.Verbose(ErrorCode.ProxyClient_ThreadAbort, "Received thread abort exception -- exiting. {0}", tae);
+                logger.Verbose(ErrorCode.ProxyClient_ThreadAbort, "Received thread abort exception -- exiting. {0}", exc);
                 Thread.ResetAbort();
                 return null;
             }
-            catch (OperationCanceledException oce)
+            catch (OperationCanceledException exc)
+            {
+                logger.Verbose(ErrorCode.ProxyClient_OperationCancelled, "Received operation cancelled exception -- exiting. {0}", exc);
+                return null;
+            }
+            catch (ObjectDisposedException exc)
+            {
+                logger.Verbose(ErrorCode.ProxyClient_ReceiveError, "Received Object Disposed exception -- exiting. {0}", exc);
+                return null;
+            }
+            catch (InvalidOperationException exc)
             {
-                logger.Verbose(ErrorCode.ProxyClient_OperationCancelled, "Received operation cancelled exception -- exiting. {0}", oce);
+                logger.Verbose(ErrorCode.ProxyClient_ReceiveError, "Received Invalid Operation exception -- exiting. {0}", exc);
```

Comment:
I believe these new exceptions should log with a different error code than the general exception case (`ErrorCode.ProxyClient_ReceiveError`). Maybe not worth creating an entirely new code, but at least maybe reuse `ErrorCode.ProxyClient_OperationCancelled`, which is logically very similar to canceling the `Take` call by passing a cancellation token, just a different implementation due to perf, but for the end user reading the log it should be no different.